lisa/node.py

```diff
@@ -283,7 +283,9 @@ def capture_system_information(self, name: str = "") -> None:
         self.log.debug(f"capturing system information to {saved_path}.")
         self.os.capture_system_information(saved_path)
 
-    def find_partition_with_freespace(self, size_in_gb: int) -> str:
+    def find_partition_with_freespace(
+        self, size_in_gb: int, use_os_disk: bool = False
```

Comment:
`use_os_disk_only`

Reply:
We do not need to necessarily constraint to the OS disk. The logic here when `use_os_disk` is:
1. Try to check if os disk can be used and has enough available space
2. Otherwise check the data disks

Reply:
If so, I think it doesn't need to be checked. I don't see any case need to skip os disk. The data disk is attached, because in some cases the os disk is not enough.

Reply:
I think your explanation makes sense. All we need to do in our test cases is to ensure that an additional data disk is available. I will make the change

Reply:
I went through the code again and we need this flag. For example, we need to pass a block device that it can modify and it can't be an OS disk

Reply:
1. All disks are "block devices". It's confusing to use this variable to differentiate os disk.
2. Because the WSL is running in a managed environment, can it be changed to pass in a device name directly? If it's not possible, let's try a clearer name.

Reply:
The variable is still block disk. See my latest replies, considering passing in the disk name. It may be more flexible.
`        use_block_disk = variables.get("use_block_disk", True)`